oxidation/libparsec/crates/types/src/regex.rs

```diff
@@ -12,43 +13,142 @@ fn escape_globing_pattern(string: &str) -> String {
+        Ok(Self(
+            raw_regexes
+                .iter()
+                .map(|l| regex::Regex::new(l).map_err(|err| RegexError::ParseError { err }))
+                .collect::<Result<Vec<regex::Regex>, RegexError>>()?,
+        ))
     }
 
-    pub fn is_match(&self, string: &str) -> bool {
-        self.0.is_match(string)
+    /// Returns a regex from a glob pattern
+    pub fn from_glob_pattern(pattern: &str) -> RegexResult {
+        let escaped_str = escape_globing_pattern(pattern);
+        Ok(Regex(vec![fnmatch_regex::glob_to_regex(&escaped_str)
+            .map_err(|err| RegexError::GlobPatternError { err })?]))
     }
-}
 
-impl AsRef<str> for Regex {
-    fn as_ref(&self) -> &str {
-        self.0.as_str()
+    pub fn from_regex_str(regex_str: &str) -> RegexResult {
+        Self::from_raw_regexes(&[regex_str])
+    }
+
+    pub fn is_match(&self, string: &str) -> bool {
+        self.0.iter().any(|r| r.is_match(string))
     }
 }
 
 impl PartialEq for Regex {
+    // This overload is only used in python tests. We are not using HashSet directly
+    // in the `Regex` type because `regex::Regex` has no implementation of `Hash`.
     fn eq(&self, other: &Self) -> bool {
-        self.as_ref() == other.as_ref()
+        if self.0.len() == other.0.len() {
+            let set: HashSet<&str> = HashSet::from_iter(self.0.iter().map(|r| r.as_str()));
+
+            other.0.iter().all(|r| set.contains(r.as_str()))
+        } else {
+            false
+        }
     }
 }
 
 impl Display for Regex {
     fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
-        self.0.fmt(f)
+        write!(
+            f,
+            "{}",
+            self.0
+                .iter()
+                .map(regex::Regex::as_str)
+                .collect::<Vec<_>>()
+                .join("|")
+        )
     }
 }
 
 #[cfg(test)]
 mod tests {
+    use std::{fs, path::Path, sync::Once};
+
     use crate::regex::Regex;
     use rstest::rstest;
 
+    const TEST_DIR_PATH: &str = "tmp-parsec-regex-tests/";
+    static INIT: Once = Once::new();
+
+    #[rstest]
+    #[case::base("*.rs\n*.py", "tmp-parsec-regex-tests/base.tmp")]
+    #[case::trim_whitespace("  *.rs\n   *.py   ", "tmp-parsec-regex-tests/trim_whitespace.tmp")]
+    #[case::empty_lines("*.rs\n\n\n*.py", "tmp-parsec-regex-tests/empty_lines.tmp")]
+    #[case::trim_whitespace_and_empty_lines(
+        "   *.rs\n\n  \n\n*.py  ",
+        "tmp-parsec-regex-tests/trim_whitespace_and_empty_lines.tmp"
+    )]
+    #[case::ignore_comment(
+        "# This contains patterns\n## yes\n   *.rs\n\n  \n\n*.py  ",
+        "tmp-parsec-regex-tests/ignore_comment.tmp"
+    )]
+    fn from_pattern_file_content(#[case] file_content: &str, #[case] filename: &str) {
+        INIT.call_once(|| {
+            if !Path::new(TEST_DIR_PATH).exists() {
+                fs::create_dir(TEST_DIR_PATH).unwrap();
+            }
+        });
```

Comment:
I was meaning a real temp dir like `/tmp` on linux

Reply:
see `std::env::temp_dir()`

Reply:
or the fixture `tmp_path` in rust

Reply:
Sorry, I thought windows has no real tmp folders so I made a directory like this. I fixed it.